test/client_ui_test.py

```diff
@@ -37,55 +37,48 @@ def test_draw():
 
     # Size is unknown at this point.
     pb = client.ProgressBar(output=f, now=fake_time)
-    assert f.last == (
-        "[ ---- ] 0 bytes, 0.00 seconds, 0 bytes/s".ljust(79) + "\r"
-    )
+    line = "[ ---- ] 0 bytes, 0.00 seconds, 0 bytes/s"
+    assert f.last == line.ljust(79) + "\r"
 
     # Size was updated, but no bytes were transfered yet.
     fake_time.now += 0.1
     pb.size = 3 * GiB
     pb.update(0)
-    assert f.last == (
-        "[   0% ] 0 bytes, 0.10 seconds, 0 bytes/s".ljust(79) + "\r"
-    )
+    line = "[   0% ] 0 bytes, 0.10 seconds, 0 bytes/s"
+    assert f.last == line.ljust(79) + "\r"
 
     # Write some data...
     fake_time.now += 1.0
     pb.update(512 * MiB)
-    assert f.last == (
-        "[  16% ] 512.00 MiB, 1.10 seconds, 465.45 MiB/s".ljust(79) + "\r"
-    )
+    line = "[  16% ] 512.00 MiB, 1.10 seconds, 465.45 MiB/s"
+    assert f.last == line.ljust(79) + "\r"
 
     # Write zeros (much faster)...
     fake_time.now += 0.2
     pb.update(2 * GiB)
-    assert f.last == (
-        "[  83% ] 2.50 GiB, 1.30 seconds, 1.92 GiB/s".ljust(79) + "\r"
-    )
+    line = "[  83% ] 2.50 GiB, 1.30 seconds, 1.92 GiB/s"
+    assert f.last == line.ljust(79) + "\r"
 
     # More data, slow down again...
     fake_time.now += 1.0
     pb.update(512 * MiB)
-    assert f.last == (
-        "[ 100% ] 3.00 GiB, 2.30 seconds, 1.30 GiB/s".ljust(79) + "\r"
-    )
+    line = "[ 100% ] 3.00 GiB, 2.30 seconds, 1.30 GiB/s"
+    assert f.last == line.ljust(79) + "\r"
 
     # Flush takes some time, lowering final rate.
     fake_time.now += 0.1
     pb.close()
-    assert f.last == (
-        "[ 100% ] 3.00 GiB, 2.40 seconds, 1.25 GiB/s".ljust(79) + "\n"
-    )
+    line = "[ 100% ] 3.00 GiB, 2.40 seconds, 1.25 GiB/s"
+    assert f.last == line.ljust(79) + "\n"
 
 
 def test_with_size():
     fake_time = FakeTime()
     f = FakeFile()
 
     client.ProgressBar(size=3 * GiB, output=f, now=fake_time)
-    assert f.last == (
-        "[   0% ] 0 bytes, 0.00 seconds, 0 bytes/s".ljust(79) + "\r"
-    )
+    line = "[   0% ] 0 bytes, 0.00 seconds, 0 bytes/s"
+    assert f.last == line.ljust(79) + "\r"
```

Comment:
Since this assertions all look the same, maybe worth extracting them to a function?
```python
def assert_last_progress_match(file, expected_line):
    assert file.last == expected_line.ljust(79) + "\r"
```

Reply:
These are not new, let’s improve this later.